plugins/couchbase/couchbaseembed.cpp

```diff
@@ -360,6 +366,323 @@ namespace couchbaseembed
+                    }
+                }
+
+                activeConnections.clear();
+            }
+
+            void releaseActive(CouchbaseConnection* connectionPtr)
+            {
+                CriticalBlock block(cacheLock);
+                bool wasFound = false;
+
+                // Find given connection in our active list and move it to our
+                // idle list
+                for (ObjMap::iterator keyIter = activeConnections.begin(); !wasFound && keyIter != activeConnections.end(); keyIter++)
+                {
+                    for (ConnectionList::iterator connectionIter = keyIter->second.begin(); !wasFound && connectionIter != keyIter->second.end(); connectionIter++)
+                    {
+                        if (*connectionIter == connectionPtr)
+                        {
+                            connectionPtr->updateTimeTouched();
+                            keyIter->second.erase(connectionIter);
+                            idleConnections[keyIter->first].push_back(connectionPtr);
+                            wasFound = true;
+
+                            if (traceLevel > 4)
+                            {
+                                DBGLOG("Couchbase: Released connection object %p", connectionPtr);
+                            }
+                        }
+                    }
+                }
+            }
+
+            void expire()
+            {
+                if (!idleConnections.empty())
+                {
+                    CriticalBlock block(cacheLock);
+
+                    time_t oldestAllowedTime = time(NULL) - OBJECT_EXPIRE_TIMEOUT_SECONDS;
+                    __int32 expireCount = 0;
+
+                    for (ObjMap::iterator keyIter = idleConnections.begin(); keyIter != idleConnections.end(); keyIter++)
+                    {
+                        ConnectionList::iterator connectionIter = keyIter->second.begin();
+
+                        while (connectionIter != keyIter->second.end())
+                        {
+                            if (*connectionIter)
+                            {
+                                if ((*connectionIter)->getTimeTouched() < oldestAllowedTime)
+                                {
+                                    delete(*connectionIter);
+                                    connectionIter =  keyIter->second.erase(connectionIter);
+                                    ++expireCount;
+                                }
+                                else
+                                {
+                                    ++connectionIter;
+                                }
+                            }
+                            else
+                            {
+                                connectionIter =  keyIter->second.erase(connectionIter);
+                            }
+                        }
+                    }
+
+                    if (traceLevel > 4 && expireCount > 0)
```

Comment:
I don't think it makes sense here. Reviewing other usage of traceLevel throughout the platform, it appears that bare numerics are used instead of enums from that class.